### PR TITLE
Twitch handler

### PR DESCRIPTION
**Pull request by luuky50**
Added the twitch api
Added the first working twitch movement system

### CODE REVIEW

File: Assets/Scripts/Class/Secrets.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+public static class Secrets
+{
+	//public const string CLIENT_ID = "gp762nuuoqcoxypju8c569th9wz7q5"; //Your application's client ID, register one at https://dev.twitch.tv/dashboard
+	public const string OAUTH_TOKEN = "5cg8jhk19tsv7o4xhegsdvifk3tgf8"; //A Twitch OAuth token which can be used to connect to the chat
+	public const string USERNAME_FROM_OAUTH_TOKEN = "robotcommander"; //The username which was used to generate the OAuth token
+	public const string CHANNEL_ID_FROM_OAUTH_TOKEN = "642538431"; //The channel Id from the account which was used to generate the OAuth token
+}
+
+//642538431
```
Comment (Sempakonka) [inline Assets/Scripts/Class/Secrets.cs]: Could you explain this documtation?

File: Assets/Scripts/RobotMovement.cs [hunk 2]
```diff
@@ -60,7 +60,7 @@ private void OnCollisionEnter(Collision col)
             ? new Vector3(0f, 0f, 90f) : new Vector3(0f, 0f, 0f);
     }
 
-    private void MoveSideways(SidewaysDirections direction)
```
Comment (Sempakonka) [inline Assets/Scripts/RobotMovement.cs:63]: All public members should be documented

File: Assets/Scripts/RobotMovement.cs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 using UnityEngine;
 
-enum SidewaysDirections
```
Comment (Sempakonka) [inline Assets/Scripts/RobotMovement.cs:3]: all public members should be documented

File: Assets/Scripts/TwitchClient.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		Debug.Log($"Message received from {e.ChatMessage.UserId}: {e.ChatMessage.Message}");
+	}
+
+	
+	private void OnChatCommandReceived(object sender, TwitchLib.Client.Events.OnChatCommandReceivedArgs e)
+	{
+		switch (e.Command.CommandText)
+		{
+			case "d":
+				//_client.SendWhisper(e.Command.ChatMessage.Username, "Its working!");
```
Comment (Sempakonka) [inline Assets/Scripts/TwitchClient.cs]: I fail to understand this documentation. Could you elaborate?

File: Assets/Scripts/TwitchClient.cs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+
+    private void Update()
+	{
+		// Don't call the client send message on every Update,
+		// this is sample on how to call the client,
+		// not an example on how to code.
+		if (Input.GetKeyDown(KeyCode.Space))
+		{
+			_client.SendWhisper(_channelToConnectTo, "Your mom");
```
Comment (Sempakonka) [inline Assets/Scripts/TwitchClient.cs]: I do not understand this action. What does the provided message "Your mom" give in terms of information?

File: Assets/Scripts/TwitchClient.cs [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+				_client.SendMessage(e.Command.ChatMessage.Channel, $"Unknown chat command: {e.Command.CommandIdentifier}{e.Command.CommandText}");
+				break;
+		}
+	}
+
+
+
+    private void Update()
+	{
+		// Don't call the client send message on every Update,
```
Comment (Sempakonka) [inline Assets/Scripts/TwitchClient.cs:114]: Please add ad //TODO to prevent forgetting things such as these

File: Assets/Scripts/TwitchPubSub.cs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+		Debug.Log("sendingTopics");
+	}
+
+private void OnWhisper(object sender, TwitchLib.PubSub.Events.OnWhisperArgs e)
+	{
+		Debug.Log("ITS GETTING A WHISPER");
+		Debug.Log($"{e.Whisper.Data}");
+		// Do your bits logic here.
+	}
+}
```
Comment (Sempakonka) [inline Assets/Scripts/TwitchPubSub.cs:69]: This code looks similiar to 'TwitchClient.cs'. What functionality does it provide to have this duplicate line of code? Could this be improved?

**Review by Sempakonka — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Class/Secrets.cs (hunk 1 above)
Comment (luuky50) [reply Assets/Scripts/Class/Secrets.cs]: This comment can be removed

File: Assets/Scripts/TwitchClient.cs (hunk 5 above)
Comment (luuky50) [reply Assets/Scripts/TwitchClient.cs]: This was a temporary debug message this will be changed

File: Assets/Scripts/TwitchClient.cs (hunk 4 above)
Comment (luuky50) [reply Assets/Scripts/TwitchClient.cs]: This will be removed, it was only a temporary debug message

File: Assets/Scripts/TwitchPubSub.cs (hunk 7 above)
Comment (luuky50) [reply Assets/Scripts/TwitchPubSub.cs:69]: We can use this script to use pubsub features from Twitch, I will investigate the full functionality of this feature when needed.
[PubSub](https://dev.twitch.tv/docs/pubsub#:~:text=PubSub%20enables%20you%20to%20subscribe,broadcast%20realtime%20messages%20to%20clients.&text=An%20instant%20messaging%20service%20sending%20instant%20messages%20between%20friends.)

**Review by Sempakonka — APPROVED** [review]

**Comment by Sempakonka** [comment]
LMTG!

**Pull request merged**